extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -253,18 +309,23 @@ export class AmpA4A extends AMP.BaseElement {
     //   - Uncaught error otherwise => don't return; percolate error up
     this.adPromise_ = viewerFor(this.win).whenFirstVisible()
         // This block returns the ad URL, if one is available.
+        /** @return {!Promise<?string>} */
         .then(() => {
           checkStillCurrent(promiseId);
+          this.lifecycleReporter_.sendPing('buildUrl');
           return this.getAdUrl();
         })
         // This block returns the (possibly empty) response to the XHR request.
+        /** @return {!Promise<?Response>} */
         .then(adUrl => {
           checkStillCurrent(promiseId);
+          this.lifecycleReporter_.sendPing('sendAdRequest');
```

Comment:
I think we could just rename this to `adRequestStart`, and remove the other `adRequestStart`, which is very close to `constructor`.